File: java/hamming/src/main/java/Hamming.java
public class Hamming {

  private final String leftStrand;
  private final String rightStrand;

  Hamming(String leftStrand, String rightStrand) {
    this.leftStrand = leftStrand == null ? "" : leftStrand.trim().toUpperCase();
    this.rightStrand = rightStrand == null ? "" : rightStrand.trim().toUpperCase();
    if (this.leftStrand.length() != this.rightStrand.length())
      throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
  }

  int getHammingDistance() {
    if (this.leftStrand.isEmpty() || this.rightStrand.isEmpty()) return 0;
    int distance = 0;
    for (int i = 0; i < this.leftStrand.length(); i++) {
      if (this.leftStrand.charAt(i) != this.rightStrand.charAt(i))
        ++distance;
    }
    return distance;
  }
}

File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
  public static int Distance(string firstStrand, string secondStrand)
  {
    if (firstStrand.Length != secondStrand.Length)
      throw new ArgumentException("firstStrand and secondStrand must be of equal length.");

    return firstStrand.Where((x, i) => x != secondStrand[i]).Count();
  }
}